pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/DockerPipelineScript.groovy

```diff
@@ -46,8 +46,8 @@ public class DockerPipelineScript extends DeclarativeAgentScript {
         LabelScript labelScript = (LabelScript) Label.DescriptorImpl.instanceForName("label", [label: targetLabel]).getScript(script)
         return labelScript.run {
             if (!Utils.withinAStage()) {
-                script.stage(StageTagsMetadata.agentSetup()) {
-                    Utils.markSyntheticStage(StageTagsMetadata.agentSetup(), StageTagsMetadata.SYNTHETIC_PRE)
+                script.stage(SyntheticStageNames.agentSetup()) {
+                    Utils.markSyntheticStage(SyntheticStageNames.agentSetup(), Utils.getSyntheticStageMetadata().pre)
```

Comment:
we could potentially implement a new pipeline step in `tags-metadata` that allows "normal" pipeline users to tag their own stages.